segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -222,10 +221,302 @@ def generate_test_label_map(size_img=(1024, 1024), num_A=100, num_B=100, num_C=1
+        region_2_perim_x, region_2_perim_y = circle_perimeter(center_2[0], center_2[1], rad, shape=size_img)
+
+        # now add the coordinates to the respective numpy array
+        ring_region_1_x = np.concatenate((ring_region_1_x, region_1_perim_x))
+        ring_region_1_y = np.concatenate((ring_region_1_y, region_1_perim_y))
+        ring_region_2_x = np.concatenate((ring_region_2_x, region_2_perim_x))
+        ring_region_2_y = np.concatenate((ring_region_2_y, region_2_perim_y))
+
+    # create a tuple for the x and y coordinates for each cell ring
+    ring_region_1 = (ring_region_1_x, ring_region_1_y)
+    ring_region_2 = (ring_region_2_x, ring_region_2_y)
+
+    return ring_region_1, ring_region_2
+
+
+def generate_two_cell_ring_jitter_coords(size_img, ring_region_1, ring_region_2,
+                                         jitter_factor, jitter_multiplier, seed):
+    """
+    This function generates the coordinates we need to set jittered signal for rings
+    around the nucleus or membrane respectively
+
+    Arguments:
+        size_img (tuple): the dimensions of the image we wish to generate
+        ring_region_1 (tuple): a tuple representing the x and y coodinates of the first cell's ring region in question
+        ring_region_2 (tuple): similar to ring_region_1 but for cell 2
+        jitter_factor (int): controls the amount of random noise we add to the cell
+        jitter_multiplier (int): an additional jitter scatter to add more jitter the further away from
+            the membrane or nucleus
+        seed (int): whether to set the random seed or not, useful for testing
+
+    Returns:
+        ring_region_1_jitter (tuple): a tuple with the x and y coordinates to assign jitter for the specified ring for cell 1
+        ring_region_2_jitter (tuple): similar to ring_region_1_jitter for ring_region_2
+    """
+
+    if seed:
+        np.random.seed(seed)
+
+    # generate some jitter for each x and y coordinate
+    # the way we do this is to generate random offset based on the jitter_factor multiplier
+    # and sampling from a uniform distribution with an additional jitter_factor and jitter_multiplier multiplier
+    ring_region_1_x_jitter = (ring_region_1[0] + np.around(np.random.uniform(-jitter_factor * jitter_multiplier, jitter_factor * jitter_multiplier, len(ring_region_1[0])))).astype(np.int16)
+    ring_region_1_y_jitter = (ring_region_1[1] + np.around(np.random.uniform(-jitter_factor * jitter_multiplier, jitter_factor * jitter_multiplier, len(ring_region_1[1])))).astype(np.int16)
+    ring_region_2_x_jitter = (ring_region_2[0] + np.around(np.random.uniform(-jitter_factor * jitter_multiplier, jitter_factor * jitter_multiplier, len(ring_region_2[0])))).astype(np.int16)
+    ring_region_2_y_jitter = (ring_region_2[1] + np.around(np.random.uniform(-jitter_factor * jitter_multiplier, jitter_factor * jitter_multiplier, len(ring_region_2[1])))).astype(np.int16)
```

Comment:
You can just call `np.random.uniform` just once with size/shape set to `ring_region_1.size + ring_region_2.size` and do:
```
num_coords1 = ring_region_1.size/2
ring_region_1_x_jitter = ring_region_1[0] + jitter[:num_coords1]
ring_region_1_y_jitter = ring_region_1[1] + jitter[num_coords1:2*num_coords1]
...
```
that should be faster and resolve pep8 inspired formatting headaches